### PR TITLE
Deprecate ClearConfiguration

### LINKED ISSUE #1767: Deprecate ClearConfiguration
As discussed in  #1764, it's somewhat confusing, and untangling it is probably not worth the effort. The preferred method of getting a Fake with no configuration is to just make a new one.

### PR DESCRIPTION
**Pull request by blairconrad**
Fixes #1767.

### CODE REVIEW

File: docs/advanced-usage.md [hunk 1]
```diff
@@ -4,16 +4,8 @@ FakeItEasy exposes a few APIs that aren't commonly needed, but can be useful in
...
-```csharp
-var foo = A.Fake<IFoo>();
-Assert.Equal(0, foo.Bar());
-A.CallTo(() => foo.Bar()).Returns(42);
-Assert.Equal(42, foo.Bar());
-Fake.ClearConfiguration(foo);
-Assert.Equal(0, foo.Bar());
-```
+The best way to remove a fake's configuration is to discard the fake and create
+a new one to replace it.
```
Comment (thomaslevesque) [inline docs/advanced-usage.md:8]: Saying that it's the "best way" suggests there's an other way. Maybe you could mention that `ClearConfiguration` exists, but is deprecated? Or just don't say that it's the best way, and don't mention `ClearConfiguration`?

**Review by thomaslevesque — CHANGES_REQUESTED** [review]
Thanks @blairconrad, looks good to me. I just made one suggestion about the doc changes.

File: docs/advanced-usage.md (hunk 1 above)
Comment (blairconrad) [reply docs/advanced-usage.md]: You're right. I've given it another go (I just amended the commit; no changes to any other files). I'm keen to hear your thoughts.

Comment (thomaslevesque) [reply docs/advanced-usage.md]: Perfect!

**Review by thomaslevesque — APPROVED** [review]

**Comment by thomaslevesque** [comment]
Thanks @blairconrad!

**Comment by blairconrad** [comment]
Thank you, @thomaslevesque. An astute comment and alacritous merging.

**Pull request merged**